inc/features/inventoriable.class.php

```diff
@@ -125,11 +125,20 @@ protected function showInventoryInfo() {
       echo '</tr>';
 
       $agent = new Agent();
-      $has_agent = $agent->getFromDBByCrit([
-         'itemtype' => $this->getType(),
-         'items_id' => $this->fields['id']
+      $iterator = $DB->request(Agent::getTable(), [
+         'WHERE'     => [
+            'itemtype' => $this->getType(),
+            'items_id' => $this->fields['id']
+         ],
+         'ORDERBY'    => ['last_contact DESC'],
```

Comment:
You could limit data returned by first request.
```suggestion
      $iterator = $DB->request([
         'SELECT'    => ['id'],
         'FROM'      => Agent::getTable(),
         'WHERE'     => [
            'itemtype' => $this->getType(),
            'items_id' => $this->fields['id']
         ],
         'ORDERBY'   => ['last_contact DESC'],
         'LIMIT'     => 1
```